src/commands/inputs.rs

```diff
@@ -0,0 +1,349 @@
+                let name = decl.name().as_str().to_string();
+                
+                // Track if it has a default
+                let has_default = decl.expr().is_some();
+                input_defaults.insert(name.clone(), has_default);
+                
+                // Check if the default is a literal and store the expression
+                if let Some(expr) = decl.expr() {
+                    let is_literal = matches!(expr, wdl::ast::v1::Expr::Literal(_));
+                    literal_defaults.insert(name.clone(), is_literal);
+                    
+                    // Store the default expression as a string
+                    default_values.insert(name, expr.syntax().to_string());
+                } else {
+                    literal_defaults.insert(name, false);
+                }
+            }
+        }
+    }
+
+    (input_defaults, literal_defaults, default_values)
+}
+
+/// Converts a WDL type to its JSON representation
+fn type_to_json(ty: &Type) -> Value {
+    match ty {
+        Type::Primitive(ty, _bool) => match ty {
+            wdl::analysis::types::PrimitiveType::Boolean => Value::String("Boolean".to_string()),
+            wdl::analysis::types::PrimitiveType::Integer => Value::String("Integer".to_string()),
+            wdl::analysis::types::PrimitiveType::Float => Value::String("Float".to_string()),
+            wdl::analysis::types::PrimitiveType::String => Value::String("String".to_string()),
+            wdl::analysis::types::PrimitiveType::File => Value::String("File".to_string()),
+            wdl::analysis::types::PrimitiveType::Directory => {
+                Value::String("Directory".to_string())
+            }
+        },
+
+        _ => Value::Null,
+    }
+}
+
+/// Collects inputs with their parent names
+fn collect_inputs_with_parents<'a>(
+    args: &'a InputsArgs,
+    document: &'a Document,
+) -> Result<Vec<(&'a str, &'a str, &'a Input)>> {
+    let mut result: Vec<(&str, &str, &Input)> = Vec::new();
+
+    if let Some(name) = &args.name {
+        // Specific task or workflow requested
+        if let Some(task) = document.task_by_name(name) {
+            // Error if nested-inputs is used with a task
+            if args.nested_inputs {
+                anyhow::bail!("--nested-inputs is only valid for workflows, not tasks");
+            }
+            
+            for (input_name, input) in task.inputs() {
+                result.push((task.name(), input_name, input));
+            }
+        } else if let Some(workflow) = document.workflow() {
+            if workflow.name() == name {
+                // Add workflow inputs
+                for (input_name, input) in workflow.inputs() {
+                    result.push((workflow.name(), input_name.as_str(), input));
+                }
+
+                // If nested_inputs is true, add all called task inputs
+                if args.nested_inputs {
+                    for task in document.tasks() {
```

Comment:
this is also incorrect. Instead of iterating through the _document's tasks_, we want to iterate through the _workflow's calls_. https://docs.rs/wdl-analysis/latest/wdl_analysis/document/struct.Workflow.html#method.calls